Capstone Project Pull Request - for Amkam DevBootcamp

AmaduKamara opened the pull request:
# Amkam DevBootcamp

## All Required ♻️ changes have been made successfully in the project

- I've done the Mobile-first approach as required
- Changed all section elements in my codebase to the article element
- Changed the div element for the top nav to nav element
- Used loop on the array of tutors and pass them to the HTML of featured tutors.

All Required changes have been made 👍
Thanks for the review. I think my code now looks great and no more repetition of code in JavaScript

I've followed all requirements in the curriculum and was able to build on a Bootcamp idea on which I modeled my design as closely as the one given in the curriculum.

## Video of the website run through

I explained the website idea and the different pages and features. Please take a look at the video to understand the idea I worked on.
<a href="https://www.loom.com/share/b93657f71d7c4ff5a9aeeb547dd7038f"> Amkam DevBootcamp Website Video </a>

## Live Demo of the website
<a href="https://amadukamara.github.io/amkam-dev-bootcamp/">Amkam DevBootcamp Live Demo</a>

AmaduKamara committed:
Setup project linters, index page, sections and styled hero main prograam sections.

AmaduKamara committed:
Completed homepage and styles for homepage and created about page

AmaduKamara committed:
Add contents on the README.md file for project documentation

AmaduKamara committed:
Add button in hero section of index.html and contact page with all styles

AmaduKamara committed:
Add styles for homepage mobile styles

AmaduKamara committed:
Add mobile styles for about page

AmaduKamara committed:
Increase font-size of about hero description

AmaduKamara committed:
Add styles for mobile menu

AmaduKamara committed:
Add JavaScript app.js file for mobile menu interraction

AmaduKamara committed:
Edit README.md

AmaduKamara committed:
Add screenshots of the website in README.md

AmaduKamara committed:
Changed screenshots in README.md

AmaduKamara committed:
Edit prohrams and give it full with on mobile

AmaduKamara committed:
Add tutors.js file for holding featured turors array of objects

AmaduKamara committed:
Add array of objects for feaatured tutors in tutors.js file

AmaduKamara committed:
Worked on parsing tutors objects data to index page and adjusted the styles of each tutor

AmaduKamara committed:
Delpoyed aapp and add demo link to README.md

AmaduKamara committed:
Edit About heading 1 to cocogoose font

AmaduKamara committed:
Give credit of the original design in README.md file

AmaduKamara committed:
Add description of the project to README.md file

AmaduKamara committed:
Updated README documentaion and styles in main.css

AmaduKamara committed:
Edit webhint errors on bad image src file

AmaduKamara committed:
Edit stylelint error on main.css file

AmaduKamara committed:
Ran stylelint --fix command to fix all css errors in main.css

AmaduKamara committed:
fixed unexpected styles in  main.css

AmaduKamara committed:
Worked on css linters and spacing

AmaduKamara committed:
Worked on css linters and spacing

AmaduKamara committed:
Fix linter errons on aabout page

AmaduKamara committed:
Fix linter errons on aabout page

AmaduKamara committed:
Re-order styles to solve stylelint errors

AmaduKamara committed:
Add cocogoose font file into project

AmaduKamara committed:
Add cocogoose font file into project

AmaduKamara committed:
Removed cocogoose font due to linter not recognizing it and add Roboto font

AmaduKamara committed:
Edit indentation error stylelint

AmaduKamara committed:
Edit backgroung-image to background

AmaduKamara committed:
Edit backgroung-image to background

AmaduKamara committed:
Edit background url

AmaduKamara committed:
Edit background-image naming

AmaduKamara committed:
Edit background-image to background

AmaduKamara committed:
Add single quote for background-images

AmaduKamara committed:
Add background-position on background-image styles

AmaduKamara committed:
Edit event-left styles

AmaduKamara committed:
Removed newline and 2 spaces indentation in main.css

AmaduKamara committed:
Edit single spacings

AmaduKamara committed:
Removed single spacings

oxenprogrammer commented on styles/main.css:
```diff
@@ -0,0 +1,792 @@
+  .logo-description .mobile-description {
+    display: block;
+    font-size: 18px;
+  }
+
+  /* Events Styles */
+  .events {
+    width: 100%;
+    display: flex;
+    flex-direction: column;
+  }
+
+  .events .event {
+    width: 90%;
+  }
+
+  /* Footer styles */
+  .about-footer {
+    width: 90%;
+    display: flex;
+  }
+
+  .about-footer div {
+    width: 50%;
+  }
+
+  .about-footer div img {
+    width: 100%;
+  }
+
+  /* CONTACT PAGE */
+  .contact-footer {
+    width: 100%;
+    padding: 40px 20px;
+  }
+
+  .footer-devbootcamp-logo {
+    width: 100%;
+  }
+}
```
**Changes Required**
- [x] Good effort on using the media query. However, the approach you have used is referred to as `Desktop-first Design`. For this project, we want to use the `Mobile-First Design` approach. we start out designing at the smallest viewport possible, which we will call a **mobile viewport**, and we then progressively add styles and sometimes other enhancements as the viewport increases using media queries.

AmaduKamara replied:
Ok, I'll see to that. I just find it time-consuming to use that approach first. But will get used to it anyways.
I don't know if one may call it a habit or what, but I'm so much used to designing for desktop first and then applying styles for Mobile. Since all end up being responsive, I just think it's the same thing.
👍

oxenprogrammer replied:
- [ ] Mobile-First approach is not just about being responsive, responsiveness is just one of the many benefits. Please check out [this article](https://highlandsolutions.com/blog/why-mobile-first-design-is-important) on what exactly Mobile-First approach is.

Emmyn5600 replied:
Well solved

oxenprogrammer commented on tutors.js:
```diff
@@ -0,0 +1,93 @@
+const brad = tutors.find((tutor) => tutor.id === '3');
+const jonas = tutors.find((tutor) => tutor.id === '4');
+const andrew = tutors.find((tutor) => tutor.id === '5');
+const bucky = tutors.find((tutor) => tutor.id === '6');
+
+// Colt Steel
+document.getElementById('speaker-colt').textContent = colt.name;
+document.getElementById('colt-position').textContent = colt.position;
+document.getElementById('colt-description').textContent = colt.description;
+document.getElementById('colt-image').src = colt.image;
+
+// Maximillian
+document.getElementById('speaker-max').textContent = maxi.name;
+document.getElementById('max-position').textContent = maxi.position;
+document.getElementById('max-description').textContent = maxi.description;
+document.getElementById('max-image').src = maxi.image;
+
+// Brad
+document.getElementById('speaker-brad').textContent = brad.name;
+document.getElementById('brad-position').textContent = brad.position;
+document.getElementById('brad-description').textContent = brad.description;
+document.getElementById('brad-image').src = brad.image;
+
+// Jonas
+document.getElementById('speaker-jonas').textContent = jonas.name;
+document.getElementById('jonas-position').textContent = jonas.position;
+document.getElementById('jonas-description').textContent = jonas.description;
+document.getElementById('jonas-image').src = jonas.image;
+
+// Andrew
+document.getElementById('speaker-andrew').textContent = andrew.name;
+document.getElementById('andrew-position').textContent = andrew.position;
+document.getElementById('andrew-description').textContent = andrew.description;
+document.getElementById('andrew-image').src = andrew.image;
+
+// Bucky
+document.getElementById('speaker-bucky').textContent = bucky.name;
+document.getElementById('bucky-position').textContent = bucky.position;
+document.getElementById('bucky-description').textContent = bucky.description;
+document.getElementById('bucky-image').src = bucky.image;
```
**Changes Required**
- [x] Good effort on storing the tutors as `Array of Objects`. However, I think we can do better, there is a lot of repetition on finding each tutor. Since the design of all the tutor cards are the same, I suggest we use `Array.map` method, iterate the tutors array and assign each tutor property to a **className** dynamically generated.
This will greatly reduce the amount of boilerplate code used.

AmaduKamara replied:
I've used loop now on the array and it totally makes my code shorter and no more repetition of my code.
That was a good review. All is working great now. Thanks

Emmyn5600 replied:
Well solved

oxenprogrammer commented on contact.html:
```diff
@@ -0,0 +1,200 @@
+    <!-- Navigation Header -->
+    <header>
+      <!-- Top and Navigation Header -->
+      <div class="bg-secondary top-header">
+        <div class="container d-flex justify-content-end">
+          <div class="d-flex">
+            <ul class="d-flex">
+              <li class="me-3"><i class="fab fa-facebook-f"></i></li>
+              <li><i class="fab fa-twitter"></i></li>
+            </ul>
+            <ul class="d-flex">
+              <li class="me-3 top-header-links">English</li>
+              <li class="me-3 top-header-links">My Page</li>
+              <li class="top-header-links">Logout</li>
+            </ul>
+          </div>
+        </div>
+      </div>
```
**Optional Change**
- [x] This could as well be a HTML5 `nav` element since its intent is to navigate away to social media using links.

AmaduKamara replied:
Okay, let me hit that out.

Emmyn5600 replied:
Well solved

oxenprogrammer commented on contact.html:
```diff
@@ -0,0 +1,200 @@
+    <section
+      class="contact-form d-flex justify-content-center align-items-center"
+    >
+      <div>
+        <h2 class="text-center">Contact Us</h2>
+        <div class="line"></div>
+
+        <form action="https://formspree.io/f/mayarpel" method="POST">
+          <div>
+            <input
+              type="text"
+              name="user_name"
+              placeholder="Enter full name"
+              required
+            />
+          </div>
+          <div>
+            <input
+              type="email"
+              name="user_email"
+              placeholder="Enter email address"
+              required
+            />
+          </div>
+          <div>
+            <textarea
+              name="user_message"
+              id="message"
+              cols="40"
+              rows="5"
+              maxlength="500"
+              required
+              placeholder="Write your message here..."
+            ></textarea>
+          </div>
+          <button type="submit">Send</button>
+        </form>
+      </div>
```
**Changes Required**
- [x] Let us please replace the immediate children of all HTML5 `section` elements in the **entire code base** with the HTML5 `article` element. The use of HTML5 semantics greatly improves our web page `accessibility`.

AmaduKamara replied:
Sure, I can do that boss. 👍

Emmyn5600 replied:
Well solved

oxenprogrammer reviewed (changes requested):
Hi @AmaduKamara, 

Good job so far!
I learned a lot from Becky as well while at the university, especially Java and PHP.
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!
**To Highlight**
- [x] Good command of UI responsive design.
- [x] Well explained video presentation.

## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

AmaduKamara committed:
Work on Mobile-first approach for all pages

AmaduKamara committed:
Completed styles for Desktop screens

AmaduKamara committed:
Finished atyling for featured tutors and implement the JavaScript part for dynamic data parsing.

AmaduKamara commented:
## Completed all required changes in the project:
- I've done the Mobile-first approach as required
- Changed all section elements in my codebase to the article element
- Changed the div element for the top nav to nav element
- Used loop on the array of tutors and pass them to the HTML of featured tutors.

All Required changes have been made 👍
Thanks for the review. I think my code now looks great and no more repetition of code in JavaScript

AmaduKamara committed:
Work on stylelint errors

AmaduKamara committed:
Edit all stylelint errors

AmaduKamara committed:
Edit all stylelint errors

AmaduKamara committed:
Edit styles reorder

Emmyn5600 commented on index.html:
```diff
@@ -0,0 +1,280 @@
+    <article
+      class="hero-section d-flex justify-content-center align-items-center"
+    >
+      <div class="container">
+        <div class="hero-intro">
+          <p>"Hello! Web Development made easy"</p>
+          <h1 class="text-uppercase">
+            Amkam DevBootcamp <br />A Place For Web Developers
+          </h1>
+        </div>
+        <div class="hero-description">
+          <p>
+            A beginner friendly programming principles and web development
+            fundamentals you ever needed to start creating stunning websites and
+            apps.
+          </p>
+        </div>
+        <div class="program-schedule">
+          <h4>2022.1.10(MON) ~ 2022.5.27(FRI)</h4>
+          <p>@231A City Road Wellington, Freetown, Sierra Leone.</p>
+          <a href="contact.html" class="enroll-button">Join us now</a>
+        </div>
+      </div>
+    </article>
```
- [x] You have implemented almost all the changes requested by the previous reviewer, however, Let us try to match with the given design. you need to change the padding-top, font-size and work on styling to meet the design provided. 

---------------Original------------------

![html15](https://user-images.githubusercontent.com/55635977/142613471-08990c1c-e025-4956-8e6c-5040cca79085.png)

---------------Yours------------------

![html16](https://user-images.githubusercontent.com/55635977/142613532-810cd72d-675e-437e-b904-a4e1b1cef56d.png)

AmaduKamara replied:
Oooh, man. That's not how it's showing on my computer here.
Let me check it out. 👍

Emmyn5600 replied:
Are you using a big screen?

AmaduKamara replied:
This is how it's showing on my computer.

![image](https://user-images.githubusercontent.com/50941074/142615617-5d459ac2-d63a-4540-afc5-24eead4ab9b6.png)

Emmyn5600 replied:
You can submit for a re-review, everything looks great on your end

AmaduKamara replied:
I have added padding in the hero section as requested.

Emmyn5600 reviewed (changes requested):
Hi, @AmaduKamara 

Good job so far! 👏🏼 

Great work on making the changes requested by the previous reviewer as well as project requirements:clap:
You've done well implementing the project requirements, but there are still some issues which aren't been addressed yet 
 
## STATUS: CHANGES REQUESTED :recycle:

### Changes requested.

- [ ] Check the comment under the review.

 #### Optional suggestions
 
 - N/A
 
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

AmaduKamara committed:
Add padding in hero section for the homepage

AmaduKamara committed:
Ran stylelint fix

AmaduKamara committed:
Edit padding for hero-intro on hero section

carlos-ssh reviewed (approved):
# 🍾🍾 **STATUS APPROVED** 👏🏽👏🏽

Hi @AmaduKamara , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



![Approved](https://user-images.githubusercontent.com/22258274/138466108-6b14c541-6a88-44c5-ba76-c64992ac1835.gif)


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
